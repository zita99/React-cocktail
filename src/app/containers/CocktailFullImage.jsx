import React from 'react';
import { hideImage } from '../actions/fullImageActions';
import { connect } from "react-redux";

export class CocktailFullImage extends React.Component {

  handleOnClickClose = event => {
    event.preventDefault();
      this.props.hideImage();
  }

  render() {

    if(!this.props.url) {
      return null
    } else {

      const ingredientsList = [];
      for (let i = 1; i <= 15; i++) {
         if (this.props.details['strIngredient' + i]) {
          //  consloe.log(this.props.details['strIngredient' + i])
           const ingredient = <li key={i}>{`${this.props.details['strIngredient' + i]} (${this.props.details['strMeasure' + i]})`}</li>;
          //  consloe.log(ingredient)
          ingredientsList.push(ingredient);
         }
      }

      return (
        <div className='fullScreen' onClick={ this.handleOnClickClose }>
          <div ><img src={ this.props.url } alt='cocktail image big' onClick={ event => event.stopPropagation()}></img></div>
          <div className='recipes-list_ingredients_detail'>
            <ul>{ ingredientsList }</ul>
            <p className='recipes-list_ingredients_heading_detail'>Instructions</p>
            <p className='recipes-list_ingredients_instructions_detail'>{ this.props.details.strInstructions }</p>
          </div>
          <button type='button' className='fullScreen-button_close' onClick={ this.handleOnClickClose }>CLOSE</button>
        </div>
      );
    }
  }
}

const mapStateToProps = (state) => {
  return {
    url: state.fullImageReducer.url,
    details: state.fullImageReducer.details
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
      hideImage: () => {
          dispatch(hideImage());
      }
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(CocktailFullImage);
