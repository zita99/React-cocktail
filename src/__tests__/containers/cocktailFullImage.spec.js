import React from 'react';
import { CocktailFullImage } from '../../app/containers/CocktailFullImage.jsx';
import { shallow, render } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('CocktailFullImage', () => {

  it('should not render if url is null', () => {

    const wrapper = shallow(
      <CocktailFullImage url={ null }/>
    );

    expect(toJson(wrapper)).toMatchSnapshot();

  });

  it('should render if url is passed', () => {

    const testUrl = 'http://cocktail-recipes-search.com/my-image.jpg';

    const wrapper = shallow(
      <CocktailFullImage url={ testUrl }/>
    );

    expect(toJson(wrapper)).toMatchSnapshot();

  });

});
