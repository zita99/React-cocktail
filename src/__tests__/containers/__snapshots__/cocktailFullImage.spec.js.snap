// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CocktailFullImage should not render if url is null 1`] = `null`;

exports[`CocktailFullImage should render if url is passed 1`] = `
<div
  className="fullScreen"
  onClick={[Function]}
>
  <div>
    <img
      alt="cocktail image big"
      onClick={[Function]}
      src="http://cocktail-recipes-search.com/my-image.jpg"
    />
  </div>
  <button
    className="fullScreen-button_close"
    onClick={[Function]}
    type="button"
  >
    CLOSE
  </button>
</div>
`;
